ocs_ci/ocs/scale_lib.py

```diff
@@ -0,0 +1,385 @@
+import logging
+import threading
+import random
+
+from tests import helpers
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources import pod, pvc, storage_cluster
+from ocs_ci.ocs import constants, cluster, machine, node
+from ocs_ci.ocs.exceptions import (
+    UnavailableResourceException, UnexpectedBehaviour, CephHealthException
+)
+
+logger = logging.getLogger(__name__)
+
+
+class FioPodScale(object):
+    """
+
+    """
+    def __init__(
+        self, kind='Pod', pod_dict_path=constants.NGINX_POD_YAML,
+        node_selector=constants.SCALE_NODE_SELECTOR
+    ):
+        """
+        Initializer function
+
+        Args:
+            kind (str): Kind of service POD or DeploymentConfig
+            pod_dict_path (yaml): Pod yaml
+            node_selector (dict): Pods will be created in this node_selector
+                eg: {'nodetype': 'app-pod'}
+
+        """
+        self._kind = kind
+        self._pod_dict_path = pod_dict_path
+        self._node_selector = node_selector
+        self._set_dc_deployment()
+        self.namespace_list = list()
+
+    @property
+    def kind(self):
+        return self._kind
+
+    @property
+    def pod_dict_path(self):
+        return self._pod_dict_path
+
+    @property
+    def node_selector(self):
+        return self._node_selector
+
+    def _set_dc_deployment(self):
+        """
+        Set dc_deployment True or False based on Kind
+        """
+        if self.kind == 'DeploymentConfig':
+            self.dc_deployment = True
+        else:
+            self.dc_deployment = False
+
+    def create_and_set_namespace(self):
+        """
+        Create and set namespace for the pods to be created
+        Create sa_name if Kind if DeploymentConfig
+        """
+        self.namespace_list.append(helpers.create_project())
+        self.namespace = self.namespace_list[-1].namespace
+        if self.dc_deployment:
+            self.sa_name = helpers.create_serviceaccount(self.namespace)
+            helpers.add_scc_policy(
+                sa_name=self.sa_name.name, namespace=self.namespace
+            )
+        else:
+            self.sa_name = None
+
+    def create_multi_pvc_pod(
+        self, pods_per_iter=5, io_runtime=3600, start_io=False
```

Comment:
is there any plan to have a check on this value vs pod/pvc limit? i.e., if `pods_per_iter` is set to 250, ensure enough number of `app-scale` nodes are available.

Reply:
Ack, I should add a condition here, env can handle this "pods_per_iter" value between 1-10 not expecting more than that. Purpose of this param is to create number of pods each iteration till scale value is reached, before each iteration start there is a check for cluster usage, iops and worker nodes availability.